### PR TITLE
feat: verify pull request

### PR DESCRIPTION
**Pull request by ramizpolic**
Adds a Github Action step that verifies PRs by checking for merge commits

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -215,3 +215,14 @@ jobs:
 
       - name: Dependency Review
         uses: actions/dependency-review-action@6c5ccdad469c9f8a2996bfecaec55a631a347034 # v3.1.0
+
+  verify:
```
Comment (sagikazarmark) [inline .github/workflows/ci.yaml:219]: As I said somewhere before, this should probably go into its own PR checks flow.

https://github.com/spf13/viper/blob/master/.github/workflows/checks.yaml

**Review by akijakya — APPROVED** [review]
